src/app/core/authentication/authentication.service.ts

```diff
@@ -1,142 +1,58 @@
+      switchMap(user => {
+        if (user) {
+          return this.afs.doc<any>(`users/${user.uid}`).valueChanges();
+        } else {
+          return of(null);
+        }
+      })
+    );
   }
 
-  public isAuthenticated(): boolean {
-    const expiresAt = JSON.parse(localStorage.getItem('expires_at') || '{}');
-    if (new Date().getTime() < expiresAt) {
-      return true;
-    } else {
-      this.logout();
-      return false;
-    }
-  }
-  public login(): void {
-    this.auth0.authorize();
-  }
+  async googleSignin() {
+    const provider = new auth.GoogleAuthProvider();
+    const credential = await this.afAuth.auth.signInWithPopup(provider);
 
-  public handleAuthentication(): void {
-    this.auth0.parseHash((err: any, authResult: any) => {
-      if (authResult && authResult.accessToken && authResult.idToken) {
-        this.handleUserDatabaseEntry(authResult);
-        window.location.hash = '';
-        this.setSession(authResult);
-        this.router.navigate(['/home']);
-      } else if (err) {
-        this.router.navigate(['/']);
-      }
-    });
+    this.userService.HandleUserData(credential);
+    return this.updateUserData(credential.user);
   }
 
-  private handleUserDatabaseEntry(authResult: any) {
-    this.apollo
-      .watchQuery<any>({
-        query: QUERY_USER_CHECK,
-        variables: {
-          email: authResult.idTokenPayload.email
-        }
-      })
-      .valueChanges.subscribe((res: any) => {
-        if (res.data.user.length === 0) {
-          this.apollo
-            .mutate<any>({
-              mutation: MUTATION_ADD_USER,
-              variables: {
-                objects: [
-                  {
-                    email: authResult.idTokenPayload.email,
-                    name: authResult.idTokenPayload.name,
-                    username: authResult.idTokenPayload.nickname,
-                    profile_pic: authResult.idTokenPayload.picture
-                  }
-                ]
-              }
-            })
-            .subscribe(data => {
-              localStorage.setItem('commentsLikedByUser', JSON.stringify([]));
-              localStorage.setItem('user_profile_pic', data.data.insert_user.returning[0].profile_pic);
-              localStorage.setItem('username', data.data.insert_user.returning[0].username);
-              localStorage.setItem('userId', data.data.insert_user.returning[0].id);
-            });
-        } else {
-          this.apollo
-            .watchQuery<any>({
-              query: QUERY_USER_LIKES,
-              variables: {
-                userId: res.data.user[0].id
-              }
-            })
-            .valueChanges.subscribe((likes: any) => {
-              const likedComments: any = [];
-              const upvotedIdeas: any = [];
-              const likedProjects: any = [];
-              const likedReplies: any = [];
+  async logout() {
+    await this.afAuth.auth.signOut();
+    localStorage.clear();
```

Comment:
We might want to use Firebase auth persistance rather
More info: https://firebase.google.com/docs/auth/web/auth-state-persistence

Reply:
Other variables are required too. IMO leave it be